### ISSUE TITLE
WebView: specific navigation condition causing - Missing Remote Object on 

### ISSUE THREAD

**Issue by jeeftor** [electron/electron#13137]
* Electron Version: `2.0.0`
* Operating System (Platform and Version): `OSX 10.13.14`, `Centos 7.4.1708` 
* Last known working Electron version:  `unknown`

### **Expected Behavior**

I've added a `<webview>` to electron and am navigating through a question pro survey at URL: https://www.questionpro.com/t/ALHMVZcLXK.  The survey itself loads multiple URLs with the same src:  

* https://www.questionpro.com/a/TakeSurvey?tt=FteQOiiLfd0%3D
* https://www.questionpro.com/a/TakeSurvey
* https://www.questionpro.com/a/TakeSurvey
* https://www.questionpro.com/a/TakeSurvey

My ultimate goal is to add an event listener to the `webview` so that I can a text string that only exists on the final page of the survey.  (For the example provided I'm using the letter **T** as my search string). 

My **expected behavior** is that each time I load a new page of data in `webview`  (regardless of actual url `src`) I expect it to print out the number of **T**'s found on the page.  

### **Actual behavior**

What is happening (as seen in attached screenshots) is that the first few times it works correctly but after the 3rd URL load we get an error:

**First URL**: https://www.questionpro.com/a/TakeSurvey?tt=FteQOiiLfd0%3D
> Detected  23  instances of the search term: T

**Second URL**: https://www.questionpro.com/a/TakeSurvey
> Detected  9  instances of the search term: T

**Second URL** (again): https://www.questionpro.com/a/TakeSurvey
> Detected  20  instances of the search term: T

**Second URL** (one more time): https://www.questionpro.com/a/TakeSurvey
> Uncaught Error: Cannot call function 'findInPage' on missing remote object 2

![screen shot 2018-06-01 at 4 15 13 pm](https://user-images.githubusercontent.com/6491743/40861426-087a3ef6-65b7-11e8-8c22-e5721daf6dee.png)

I have tested this as well with some other `webview` functions like `setZoomFactor` and it also errors out here.  

**To Reproduce**

If you provide a URL, please list the commands required to clone/setup/run your repo e.g.
```sh
$ git clone https://github.com/jeeftor/electronBug.git
$ cd electronBug
$ npm install
$ npm start || electron .
```

**Screenshots**
![1](https://user-images.githubusercontent.com/6491743/40860871-e60b334a-65b4-11e8-9b3a-f93c53fbaf6d.png)
![2](https://user-images.githubusercontent.com/6491743/40860873-e625bf58-65b4-11e8-9c05-e45acd8edbc0.png)
![3](https://user-images.githubusercontent.com/6491743/40860874-e637e61a-65b4-11e8-8ee7-c0c4c4e192a4.png)

**Additional Information**
# INDEX.HTML

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Hello World!</title>
  </head>
  <body>

    <webview id="webview" src="https://www.questionpro.com/t/ALHMVZcLXK" style="width:100vw; height:100vh"></webview>

    <script>
    // You can also require other files to run in this process
    require('./renderer.js')

    onload = () => {
        const webview = document.querySelector('webview')
        webview.addEventListener('load-commit', (url, isMainFrame) => {
            console.warn(`load-commit: ${url.url}`)
        })
        webview.addEventListener('did-start-loading', () => {
            console.warn('did-start-loading')
        })
        webview.addEventListener('did-stop-loading', () => {
            console.warn('did-stop-loading')
            // webview.setZoomFactor(2)
            webview.findInPage('T')
        })
        webview.addEventListener('will-navigate', (url) => {
            console.warn(`will-navigate`)
        })
        webview.addEventListener('did-navigate', (url) => {
            console.warn(`did-navigate: ${url.url}`)
        })
        webview.addEventListener('did-navigate-in-page', (isMainFrame, url) => {
            console.warn('did-navigate-in-page')
        })

        webview.addEventListener('found-in-page', (e) => {
            console.log('Detected ', e.result.matches, ' instances of the search term: T')
            webview.stopFindInPage('clearSelection')
        })
    }
    </script>
  </body>
</html>

```

**Comment by damoqiongqiu** [comment]
Same issue here, appears randomly when the app runs for a long time.

**Comment by damoqiongqiu** [comment]
Finally, I think  I have found the reason for this issue.

If we are using [ipcRenderer ](https://electronjs.org/docs/api/ipc-renderer) inside a Renderer thread,  including webview tag.

When renderer navigated to other URL, some callbacks like setTimeout of setInterval may invoke the methods on that renderer or webview instance after a period of time.

This will definitely cause "missing remote object ..."

So, the problem here is, how can we prevent ourselves from using ipcRenderer in some async callbacks?

**Comment by damoqiongqiu** [comment]
After several days of testing, I found this issue is not existing in V 1.8.x, but reappears in V 2.x.

**Comment by jeeftor** [comment]
Then I must be a feature not a bug

**Comment by mailtokartik1** [comment]
Hi I have been facing the same problem in my browser application made via Electron. I checked the behaviour by clicking the reload button many times and I receive the same error.
<img width="792" alt="screen shot 2018-06-21 at 3 43 04 pm" src="https://user-images.githubusercontent.com/29715398/41756618-6729f506-7618-11e8-93f4-ab444b763526.png">
I think the error occurs due to setTimeout function.
FYI, this issue is non existent in electron v1.8.x but it started occuring when I upgraded the electron version in my application to v2.0.0

**Comment by mailtokartik1** [comment]
Can somebody kindly update me on this issue?

**Comment by mailtokartik1** [comment]
@damoqiongqiu did you find anything?

**Comment by damoqiongqiu** [comment]
@mailtokartik1 Sorry, I didn't find anything, I am using V1.8.0 now, and everything is fine.

**Comment by mailtokartik1** [comment]
I reverted back this PR in my v2.0.0 and the missing remote error went away

**Comment by zcbenz** [comment]
@MarshallOfSound It looks like this issue and #8205 are caused by #12342, I'm not sure of the reason at first glance.

**Comment by sethlu** [comment]
I've made some comments last night in https://github.com/electron/electron/issues/8205#issuecomment-403273153 that addresses this issue in details.

**Comment by mailtokartik1** [comment]
This has been fixed in v2.0.5 I checked
